### DISCUSSION TITLE
#3573

### DISCUSSION THREAD

**Discussion by wuliaodexiaoluo** [Meloong-Git/PCL#3678] [讨论]
#3573
[错误报告-2024-4-13_13.47.12.zip](https://github.com/Hex-Dragon/PCL2/files/14965441/-2024-4-13_13.47.12.zip)

**Comment by burningtnt** [comment]
我注意到了不正常的表现：（由 HMCL 测试）

在我这边，cp 里加载的文件是 `OptiFine-1.20.4_HD_U_I8_pre4-installer.jar`，但你那边是 `OptiFine-1.20.4_HD_U_I7`。为什么一边是 -installer，另一边不是？

请你测试一下 pre4 版本

**Reply by wuliaodexiaoluo** [reply]
PCL 自动补全就是这个文件名。。。。

**Reply by burningtnt** [reply]
你把 OptiFine 改成什么样子了…… 怎么我这里无论用 installer 版本还是普通版本塞 cp 都不会有问题

**Reply by burningtnt** [reply]
此外，你上传的错误报告怎么“游戏崩溃前的输出.txt”行数这么少…… 日志都不全的

**Reply by burningtnt** [reply]
还有一个点：HMCL 启动的 Minecraft 不带有 launchwrapper-of.jar，而 PCL2 有…… （？）

**Reply by wuliaodexiaoluo** [reply]
zulu 和 Mojang JDK 都不太行的样子
那个 LaunchWrapper 应该是 oolloo 的 Java Launch Warapper （jlw）

**Reply by burningtnt** [reply]
并不是，我指的是 cp 里的 launchwrapper-of.jar，不是 JLW

**Reply by wuliaodexiaoluo** [reply]
Forge 应该是在加载 OptiFineTransformationService.transformers 的时候爆炸了
另外 Forge 49.0.39 同样不能加载 OptiFine，可见 Forge 貌似没有修复这个问题的想法

**Reply by burningtnt** [reply]
Forge 49.0.39 加载不了 OptiFine 只是因为我们通过了 cp 加载。你丢到 mods 里就没问题了。相关的 Issue 我刚刚才给 Forge 提，当然没修……

**Reply by burningtnt** [reply]
麻烦请你导出启动脚本并修改来拿到完整的控制台输出。你给的错误报告里，控制台输出不完整

**Reply by wuliaodexiaoluo** [reply]
> 此外，你上传的错误报告怎么“游戏崩溃前的输出.txt”行数这么少…… 日志都不全的

PCL 导出的是 latest.log

[debug.log](https://github.com/Hex-Dragon/PCL2/files/14965901/debug.log)
我用工具在 Java 调试模式下导出的日志
长达 288 KB

**Reply by wuliaodexiaoluo** [reply]
[错误报告-2024-4-13_17.02.27.zip](https://github.com/Hex-Dragon/PCL2/files/14965907/-2024-4-13_17.02.27.zip)

Fabric 问题依旧

**Reply by burningtnt** [reply]
> PCL 导出的是 latest.log

`游戏崩溃前的输出.txt` 是控制台输出，而不是 latest.log

**Reply by burningtnt** [reply]
你再试一次，不删除 mod.toml 直接启动，症状是启动游戏后才在 GUI 上显示错误吗？

**Reply by burningtnt** [reply]
蛮奇怪的。我猜测你删除 `mod.toml` 的时候删出问题了…… 我刚刚用 PCL2 测试了，没问题

**Comment by zkitefly** [comment]
不理解为什么会失败，你是哪里操作错了吗

### Forge 自动安装，OptiFine 做模组

成功启动游戏并加载 OptiFine

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/4e10e9a0-e95b-4cc8-a1e1-172e1cff9e5f" width="300"/>

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/d6bf8ed7-111f-4b2a-8f45-4f987ce5b511" width="300"/>

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/3e6782ce-5525-40b7-a97f-4684acd02f16" width="300"/>

### Forge OptiFine 自动安装

```
net.minecraftforge.fml.ModLoadingException: The Mod File /home/zkitefly/.minecraft/libraries/optifine/OptiFine/1.20.4_HD_U_I8_pre4/OptiFine-1.20.4_HD_U_I8_pre4-installer.jar has mods that were not found
```

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/05107682-2e2b-4161-8c9b-c85a2fcf3ca1" width="300"/>

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/3845a22f-14f7-4d2b-a018-e5cd062b97c0" width="350"/>

### Forge OptiFine 自动安装，将报错文件的 `META-INF/mods.toml` 删除

成功启动游戏并加载 OptiFine

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/09f2c267-14c5-4782-b20c-dd4f97536461" width="650"/>

<image src="https://github.com/Hex-Dragon/PCL2/assets/64117916/874df178-ac28-453f-b345-5e60eb4a5eb9" width="300"/>

**Reply by burningtnt** [reply]
如你所见~

**Comment by allMagicNB** [comment]
不理解为什么会失败，你是哪里操作错了吗

### Forge 自动安装，OptiFine 做模组

成功启动游戏并加载 OptiFine

<img src="https://github.com/Hex-Dragon/PCL2/assets/116283787/3c4a2311-ed65-4c33-b1e1-824f1df2d36b" width="300"/>

<img src="https://github.com/Hex-Dragon/PCL2/assets/116283787/90f0f60c-781c-43bb-97fb-360fef0bda10" width="300"/>

<img src="https://github.com/Hex-Dragon/PCL2/assets/116283787/e4be47b0-849e-4408-a057-7ab32bbae558" width="300"/>

### Forge OptiFine 自动安装

```
net.minecraftforge.fml.ModLoadingException: The Mod File D:\Minecraft\.minecraft\libraries\optifine\OptiFine\1.20.4_HD_U_I8_pre4\OptiFine-1.20.4_HD_U_I8_pre4.jar has mods that were not found
```
<image src="https://github.com/Hex-Dragon/PCL2/assets/116283787/8b03c307-01f3-48bd-b1ba-6211d37c33de" width="300"/>

<image src="https://github.com/Hex-Dragon/PCL2/assets/116283787/77b8e418-a1e0-4508-a7f3-141933864216" width="350"/>

### Forge OptiFine 自动安装，将报错文件的 `META-INF/mods.toml` 删除

成功启动游戏并加载 OptiFine

<img src="https://github.com/Hex-Dragon/PCL2/assets/116283787/521434e4-3d80-46a5-8273-aedf4a72d034" width="650"/>

<img src="https://github.com/Hex-Dragon/PCL2/assets/116283787/c06ff640-4ac1-475c-811c-eef5a1f5b0d9" width="300"/>

**Reply by burningtnt** [reply]
等等，游戏能跑，PCL2 自己的模组列表爆炸了？

**Reply by allMagicNB** [reply]
我看到了，我也感到奇怪，还想着要不要交 Issue。
Edit:刷新之后已经好了

**Comment by LTCatt** [comment]
我还没仔细看，但注意受到 #3657 影响，OptiFine 下载的时候可能失败，这可能会影响复现）

**Comment by wuliaodexiaoluo** [comment]
总的来说是不删 GUI 报错，删了 Minecraft 连窗口都没有直接爆炸
按道理来说用 jar 命令解压再删除然后 jar 命令重新打包不可能出问题啊？

换了个计算尝试
命令行完整操作:
```
J:\Minecraft\PCL\java\Java17-Hotspot\bin\jar xf .\OptiFine-1.20.4_HD_U_I7.jar
del J:\Minecraft\.minecraft\libraries\optifine\OptiFine\1.20.4_HD_U_I7\META-INF\mods.toml
J:\Minecraft\PCL\java\Java17-Hotspot\bin\jar cf  OptiFine-1.20.4_HD_U_I7.jar .
```
全部操作没有使用任何第三方软件完成（包括资源管理器），仅使用 cmd，del 命令，jar 命令
情况与之前完全一致，甚至日志都是一样的，完成操作后 Zulu OpenJDK、Mojang OpenJDK、Adoptium OpenJDK 均无法正常加载游戏，可以说如果这项更新被应用下来且选择删除 mods.toml ，我之后就没办法正常玩通过 PCL 安装的 OptiFine+ModLoader 的版本了
如果可以的话，这项更改我希望暂缓处理

**Reply by wuliaodexiaoluo** [reply]
[optifine.zip](https://github.com/Hex-Dragon/PCL2/files/14970665/optifine.zip)
这是经过我修改的 optifine

**Reply by wuliaodexiaoluo** [reply]
这个 optifine 甚至没有办法通过 mods 文件夹的方式被加载

**Reply by hejiehao** [reply]
~~怎么不是 OpenJ9~~

**Reply by wuliaodexiaoluo** [reply]
我这只有高版本的 JRE，PCL 用不了

**Reply by zkitefly** [reply]
不解压打包，试着用压缩包管理器（如 7z 等）找到文件选择删除看看？

**Reply by wuliaodexiaoluo** [reply]
试过没差别
我用的 Bandizip

**Comment by wuliaodexiaoluo** [comment]
<details>

<summary>完整启动参数</summary>

"J:\Minecraft\PCL\java\Java17-Hotspot\bin\java.exe" -XX:+UseG1GC -XX:-UseAdaptiveSizePolicy -XX:-OmitStackTraceInFastThrow -Dfml.ignoreInvalidMinecraftCertificates=True -Dfml.ignorePatchDiscrepancies=True -Dlog4j2.formatMsgNoLookups=true -Dfile.encoding=utf-8 -XX:MaxDirectMemorySize=256M -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Djava.library.path="J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7\1.20.4-Forge_49.0.41-OptiFine_I7-natives" -Djna.tmpdir="J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7\1.20.4-Forge_49.0.41-OptiFine_I7-natives" -Dorg.lwjgl.system.SharedLibraryExtractPath="J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7\1.20.4-Forge_49.0.41-OptiFine_I7-natives" -Dio.netty.native.workdir="J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7\1.20.4-Forge_49.0.41-OptiFine_I7-natives" -Dminecraft.launcher.brand=PCL -Dminecraft.launcher.version=320 -cp "J:\Minecraft\.minecraft\libraries\com\github\oshi\oshi-core\6.4.5\oshi-core-6.4.5.jar;J:\Minecraft\.minecraft\libraries\com\google\code\gson\gson\2.10.1\gson-2.10.1.jar;J:\Minecraft\.minecraft\libraries\com\google\guava\failureaccess\1.0.1\failureaccess-1.0.1.jar;J:\Minecraft\.minecraft\libraries\com\google\guava\guava\32.1.2-jre\guava-32.1.2-jre.jar;J:\Minecraft\.minecraft\libraries\com\ibm\icu\icu4j\73.2\icu4j-73.2.jar;J:\Minecraft\.minecraft\libraries\com\mojang\authlib\6.0.52\authlib-6.0.52.jar;J:\Minecraft\.minecraft\libraries\com\mojang\blocklist\1.0.10\blocklist-1.0.10.jar;J:\Minecraft\.minecraft\libraries\com\mojang\brigadier\1.2.9\brigadier-1.2.9.jar;J:\Minecraft\.minecraft\libraries\com\mojang\datafixerupper\6.0.8\datafixerupper-6.0.8.jar;J:\Minecraft\.minecraft\libraries\com\mojang\logging\1.1.1\logging-1.1.1.jar;J:\Minecraft\.minecraft\libraries\com\mojang\patchy\2.2.10\patchy-2.2.10.jar;J:\Minecraft\.minecraft\libraries\com\mojang\text2speech\1.17.9\text2speech-1.17.9.jar;J:\Minecraft\.minecraft\libraries\commons-codec\commons-codec\1.16.0\commons-codec-1.16.0.jar;J:\Minecraft\.minecraft\libraries\commons-io\commons-io\2.13.0\commons-io-2.13.0.jar;J:\Minecraft\.minecraft\libraries\commons-logging\commons-logging\1.2\commons-logging-1.2.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-buffer\4.1.97.Final\netty-buffer-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-codec\4.1.97.Final\netty-codec-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-common\4.1.97.Final\netty-common-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-handler\4.1.97.Final\netty-handler-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-resolver\4.1.97.Final\netty-resolver-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-transport-classes-epoll\4.1.97.Final\netty-transport-classes-epoll-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-transport-native-unix-common\4.1.97.Final\netty-transport-native-unix-common-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\io\netty\netty-transport\4.1.97.Final\netty-transport-4.1.97.Final.jar;J:\Minecraft\.minecraft\libraries\it\unimi\dsi\fastutil\8.5.12\fastutil-8.5.12.jar;J:\Minecraft\.minecraft\libraries\net\java\dev\jna\jna-platform\5.13.0\jna-platform-5.13.0.jar;J:\Minecraft\.minecraft\libraries\net\java\dev\jna\jna\5.13.0\jna-5.13.0.jar;J:\Minecraft\.minecraft\libraries\net\sf\jopt-simple\jopt-simple\5.0.4\jopt-simple-5.0.4.jar;J:\Minecraft\.minecraft\libraries\org\apache\commons\commons-compress\1.22\commons-compress-1.22.jar;J:\Minecraft\.minecraft\libraries\org\apache\commons\commons-lang3\3.13.0\commons-lang3-3.13.0.jar;J:\Minecraft\.minecraft\libraries\org\apache\httpcomponents\httpclient\4.5.13\httpclient-4.5.13.jar;J:\Minecraft\.minecraft\libraries\org\apache\httpcomponents\httpcore\4.4.16\httpcore-4.4.16.jar;J:\Minecraft\.minecraft\libraries\org\apache\logging\log4j\log4j-api\2.19.0\log4j-api-2.19.0.jar;J:\Minecraft\.minecraft\libraries\org\apache\logging\log4j\log4j-core\2.19.0\log4j-core-2.19.0.jar;J:\Minecraft\.minecraft\libraries\org\apache\logging\log4j\log4j-slf4j2-impl\2.19.0\log4j-slf4j2-impl-2.19.0.jar;J:\Minecraft\.minecraft\libraries\org\joml\joml\1.10.5\joml-1.10.5.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-glfw\3.3.2\lwjgl-glfw-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-glfw\3.3.2\lwjgl-glfw-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-glfw\3.3.2\lwjgl-glfw-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-glfw\3.3.2\lwjgl-glfw-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-jemalloc\3.3.2\lwjgl-jemalloc-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-jemalloc\3.3.2\lwjgl-jemalloc-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-jemalloc\3.3.2\lwjgl-jemalloc-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-jemalloc\3.3.2\lwjgl-jemalloc-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-openal\3.3.2\lwjgl-openal-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-openal\3.3.2\lwjgl-openal-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-openal\3.3.2\lwjgl-openal-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-openal\3.3.2\lwjgl-openal-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-opengl\3.3.2\lwjgl-opengl-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-opengl\3.3.2\lwjgl-opengl-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-opengl\3.3.2\lwjgl-opengl-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-opengl\3.3.2\lwjgl-opengl-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-stb\3.3.2\lwjgl-stb-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-stb\3.3.2\lwjgl-stb-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-stb\3.3.2\lwjgl-stb-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-stb\3.3.2\lwjgl-stb-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-tinyfd\3.3.2\lwjgl-tinyfd-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-tinyfd\3.3.2\lwjgl-tinyfd-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-tinyfd\3.3.2\lwjgl-tinyfd-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl-tinyfd\3.3.2\lwjgl-tinyfd-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl\3.3.2\lwjgl-3.3.2.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl\3.3.2\lwjgl-3.3.2-natives-windows.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl\3.3.2\lwjgl-3.3.2-natives-windows-arm64.jar;J:\Minecraft\.minecraft\libraries\org\lwjgl\lwjgl\3.3.2\lwjgl-3.3.2-natives-windows-x86.jar;J:\Minecraft\.minecraft\libraries\org\slf4j\slf4j-api\2.0.7\slf4j-api-2.0.7.jar;J:\Minecraft\.minecraft\libraries\optifine\launchwrapper-of\2.3\launchwrapper-of-2.3.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\forge\1.20.4-49.0.41\forge-1.20.4-49.0.41-universal.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\forge\1.20.4-49.0.41\forge-1.20.4-49.0.41-client.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\JarJarFileSystems\0.3.26\JarJarFileSystems-0.3.26.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\securemodules\2.2.10\securemodules-2.2.10.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\unsafe\0.9.2\unsafe-0.9.2.jar;J:\Minecraft\.minecraft\libraries\org\ow2\asm\asm\9.6\asm-9.6.jar;J:\Minecraft\.minecraft\libraries\org\ow2\asm\asm-tree\9.6\asm-tree-9.6.jar;J:\Minecraft\.minecraft\libraries\org\ow2\asm\asm-util\9.6\asm-util-9.6.jar;J:\Minecraft\.minecraft\libraries\org\ow2\asm\asm-commons\9.6\asm-commons-9.6.jar;J:\Minecraft\.minecraft\libraries\org\ow2\asm\asm-analysis\9.6\asm-analysis-9.6.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\bootstrap\2.1.0\bootstrap-2.1.0.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\bootstrap-api\2.1.0\bootstrap-api-2.1.0.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\accesstransformers\8.1.1\accesstransformers-8.1.1.jar;J:\Minecraft\.minecraft\libraries\org\antlr\antlr4-runtime\4.9.1\antlr4-runtime-4.9.1.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\eventbus\6.2.0\eventbus-6.2.0.jar;J:\Minecraft\.minecraft\libraries\net\jodah\typetools\0.6.3\typetools-0.6.3.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\forgespi\7.1.3\forgespi-7.1.3.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\coremods\5.1.0\coremods-5.1.0.jar;J:\Minecraft\.minecraft\libraries\org\openjdk\nashorn\nashorn-core\15.3\nashorn-core-15.3.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\modlauncher\10.1.2\modlauncher-10.1.2.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\mergetool-api\1.0\mergetool-api-1.0.jar;J:\Minecraft\.minecraft\libraries\com\electronwill\night-config\toml\3.6.7\toml-3.6.7.jar;J:\Minecraft\.minecraft\libraries\com\electronwill\night-config\core\3.6.7\core-3.6.7.jar;J:\Minecraft\.minecraft\libraries\org\apache\maven\maven-artifact\3.8.5\maven-artifact-3.8.5.jar;J:\Minecraft\.minecraft\libraries\net\minecrell\terminalconsoleappender\1.2.0\terminalconsoleappender-1.2.0.jar;J:\Minecraft\.minecraft\libraries\org\jline\jline-reader\3.12.1\jline-reader-3.12.1.jar;J:\Minecraft\.minecraft\libraries\org\jline\jline-terminal\3.12.1\jline-terminal-3.12.1.jar;J:\Minecraft\.minecraft\libraries\org\jline\jline-terminal-jna\3.12.1\jline-terminal-jna-3.12.1.jar;J:\Minecraft\.minecraft\libraries\org\spongepowered\mixin\0.8.5\mixin-0.8.5.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\JarJarSelector\0.3.26\JarJarSelector-0.3.26.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\JarJarMetadata\0.3.26\JarJarMetadata-0.3.26.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\fmlcore\1.20.4-49.0.41\fmlcore-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\fmlloader\1.20.4-49.0.41\fmlloader-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\fmlearlydisplay\1.20.4-49.0.41\fmlearlydisplay-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\javafmllanguage\1.20.4-49.0.41\javafmllanguage-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\lowcodelanguage\1.20.4-49.0.41\lowcodelanguage-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\libraries\optifine\OptiFine\1.20.4_HD_U_I7\OptiFine-1.20.4_HD_U_I7.jar;J:\Minecraft\.minecraft\libraries\net\minecraftforge\mclanguage\1.20.4-49.0.41\mclanguage-1.20.4-49.0.41.jar;J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7\1.20.4-Forge_49.0.41-OptiFine_I7.jar" -Djava.net.preferIPv6Addresses=system -Xmn256m -Xmx3072m -Dlog4j2.formatMsgNoLookups=true --add-exports cpw.mods.bootstraplauncher/cpw.mods.bootstraplauncher=ALL-UNNAMED -Doolloo.jlw.tmpdir="C:\ProgramData\PCL" -jar "C:\ProgramData\PCL\JavaWrapper.jar" net.minecraftforge.bootstrap.ForgeBootstrap --username wuliaodexiaoluo --version 1.20.4-Forge_49.0.41-OptiFine_I7 --gameDir "J:\Minecraft\.minecraft\versions\1.20.4-Forge_49.0.41-OptiFine_I7" --assetsDir "J:\Minecraft\.minecraft\assets" --assetIndex 12 --uuid 3XXXXXXXXXXXXXXXXXXXXXXXXXX0 --accessToken eyJraXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXsmWzs --clientId ${clientid} --xuid ${auth_xuid} --userType msa --versionType PCL --width 854 --height 480 --launchTarget forge_client --tweakClass optifine.OptiFineForgeTweaker 

</details>

**Reply by wuliaodexiaoluo** [reply]
我完全不知道我到底哪里错了
~我大概是犯了 jvm 的天条~

**Comment by burningtnt** [comment]
你原来下载的 OptiFine 就不标准吧…… 你去 optifine.net/downloads 下

**Reply by allMagicNB** [reply]
去#3692
~~甚至单独开 Discussion 了~~

**Reply by wuliaodexiaoluo** [reply]
测完了，PCL 自动补全的没办法删 mods.toml
